### DISCUSSION TITLE
How @bob is resolved into his Sphere's CID?

### DISCUSSION THREAD

**Discussion by YodaDefi** [subconsciousnetwork/noosphere#42] [Q&A] [answered]
Hello dear Subconscious team! Really excited to see what you are building.
I've listened to what you said on Youtube and read articles about the design of Noosphere.

However, i didn't get how you convert @bob -> PubKey -> CID _if @bob is not in the current user's Address Book_ . Here is the full alg as i understood it:

User has the last *CID of My Sphere*.
User wants to get contents of the "@bob/cat-thoughts" memo.

1. My Sphere is downloaded by *CID of My Sphere* from the IPFS -> *My Sphere*
2. Name "@bob" is looked up in *My Sphere's* address book -> *PubKey*
3. If *PubKey* not found in the address book -> continue (???)
4. If *PubKey* found -> **search for the sphere CID by *PubKey*** in the Noosphere Naming System (libp2p DHT) -> *User2 Sphere CID*
5. *User2 Sphere CID* is retrieved from IPFS -> *User2 Shpere*
6. Memo "cat-thoughts" is looked up in the *User2 Sphere* links -> *Cid*
7. If *Cid* is not found -> ERROR
8. If *Cid* is found -> contents is downloaded from the IPFS directly.

I didn't get what happens during step 3. Please help :-) 

---
Also, question #2: 
As long as the Address Book is "hyperlocal", it means that it is possible to have 2 Bobs in the network, right?
First one would be as seen by say User1, and another one would be as seen by say User532. Each will resolve to different PubKeys.
Am i right?

Thank you

**Comment by cdata** [comment, marked as answer]
Hey @YodaDefi thanks for watching the video, and for asking questions!

The most correct answer to your question is: undefined behavior (for now). The most likely thing we will do is: the link simply doesn't go anywhere (vaguely the equivalent of a 404 on the hypertext web).

In other words: you need a name to occur in your address book in order to resolve a public key, and you need a public key in order to find the sphere of another user.

> As long as the Address Book is "hyperlocal", it means that it is possible to have 2 Bobs in the network, right?

Yes, and this is by design of the "pet name" system ([additional reading](http://www.skyhunter.com/marcs/petnames/IntroPetNames.html)).

However, we also intend to demonstrate (eventually) how to bridge to global namespaces using this system. We will demonstrate bridging to both DNS and ENS, and probably others. To sketch this out a little bit: you'll be able to add an entry like "eth" to your address book that resolves to the public key of a bridge. By sub-addressing the bridge's address book, you can address into the ENS namespace. So, `cdata.eth` would recursively resolve from the bridge to a DID that is associated with `cdata` in ENS (and in ENS, there can be only one `cdata`).

**Reply by YodaDefi** [reply]
Thank you very much!
